### PR TITLE
feat: move over the rest of the style guide

### LINKED ISSUE #2: finish moving over the rest of the current samples guide

### PR DESCRIPTION
**Pull request by kurtisvg**
Fixes #2

### CODE REVIEW

File: content/en/_index.md [hunk 1]
```diff
@@ -212,10 +218,77 @@ requests or if they are thread-safe).
...
-// TODO
+### Coverage
+Code snippets should have reasonable test coverage and all critical code paths
+should have integration tests that test against the production service. 
 
 ## Additional best practices
 
-// TODO: Language specific? 
\ No newline at end of file
+### Adhere to language idioms
```
Comment (grayside) [inline content/en/_index.md:282]: This is one of the "guidelines" that I think should be converted to a principle.

File: content/en/_index.md [hunk 2]
```diff
@@ -212,10 +218,77 @@ requests or if they are thread-safe).
...
+* Does not directly rely on POSIX system calls
+* Handles code-sensitivity differences
+* Avoids platform-specific APIs
+
 ## Testing
 
-// TODO
+### Coverage
+Code snippets should have reasonable test coverage and all critical code paths
+should have integration tests that test against the production service.
```
Comment (grayside) [inline content/en/_index.md:278]: Not sure it fits here, I think every sample should be expected to have at least one integration or end-to-end test. The language here leaves a lot of room.

**Review by grayside — APPROVED** [review]
Anything moving forward is good, but I made a bunch of small suggestions around typos, conciseness, and clarity, and left some other structural feedback.

File: content/en/_index.md [hunk 3]
```diff
@@ -86,6 +86,38 @@ the sample work:
  {{< /tab >}}
 {{< /tabpane >}}
 
+### Method Structure
```
[the remark this answers is not in the document]
Comment (kurtisvg) [reply content/en/_index.md:89]: I split this into "Main runner" and "Minimal arguments" and "Process the result". PTAL

File: content/en/_index.md [hunk 4]
```diff
@@ -134,6 +166,13 @@ more approachable to beginners:
  {{< /tab >}}
 {{< /tabpane >}}
 
+### No CLIs
+
+Do not include CLIs for running your sample. We expect that most developers will
+copy and pase code directly from cloud.google.com into their own enviroment.
+Adding CLIs has historically been expensive to maintain in the past, and
+detracts from the purpose of the snippet itself.
```
[the remark this answers is not in the document]
Comment (kurtisvg) [reply content/en/_index.md:174]: I expect this one to be somewhat controversial, so I'm okay with it being an extra sentence if it helps defuse an escalation.

File: content/en/_index.md (hunk 1 above)
Comment (kurtisvg) [reply content/en/_index.md:282]: So we do have a principle for this in the introduction - but I think this has some necessary (or at least useful?) clarifications that idiomatic is preferred over standardize between samples.

However I'm open to removing this if you prefer.

File: content/en/_index.md (hunk 2 above)
Comment (kurtisvg) [reply content/en/_index.md:278]: It feels like a minimum of one test would be required for a "critical code path". Do you want that stated more clearly or is there something else you want?

File: content/en/_index.md [hunk 5]
```diff
@@ -134,45 +166,73 @@ more approachable to beginners:
...
-### Method Structure
-Method arguments should be limited to what is absolutely required for testing.
-In most cases, this is project specific information or the path to an external
-file. For example, project specific information (such as `projectId`) or a
-`filePath` for an external file is acceptable, while an argument for the type of
-a file or a specific action is not.
- 
+### Main runner
+
 Any declared function arguments should include a no-arg, main method with
```
[the remark this answers is not in the document]
Comment (kurtisvg) [reply content/en/_index.md:188]: Removed no-arg. I'm assuming that main methods always need to have the same arguments.

File: content/en/_index.md [hunk 6]
```diff
@@ -134,45 +166,73 @@ more approachable to beginners:
...
  documentation that lists available options (and when to use them).
  
-### Method Structure
-Method arguments should be limited to what is absolutely required for testing.
-In most cases, this is project specific information or the path to an external
-file. For example, project specific information (such as `projectId`) or a
-`filePath` for an external file is acceptable, while an argument for the type of
-a file or a specific action is not.
- 
+### Main runner
```
[the remark this answers is not in the document]
Comment (kurtisvg) [reply content/en/_index.md:186]: This is something that I adapted slightly from Java and Python, but I think makes sense across languages (particularly scripting). 

Essentially, for users using something like Vim, it's not always easy to run a specific method. Including a main that they can edit makes it streamlined to actually run it.

**Pull request merged**